File: python code/binary image processing/closing.py
import cv2

# read image
image = cv2.imread("../../images/closing.png", cv2.IMREAD_GRAYSCALE)

# initialize number of iteration
iterations = 1

# get structuring kernel
closingSize = 10
element = cv2.getStructuringElement(shape=cv2.MORPH_ELLIPSE,
                                    ksize=(2 * closingSize + 1, 2 * closingSize + 1),
                                    anchor=(closingSize, closingSize))

# apply morphological closing
morphClosingImage = cv2.morphologyEx(src=image, op=cv2.MORPH_CLOSE, kernel=element, iterations=iterations)

# display images
cv2.imshow("Original Image", image)
cv2.imshow("Closing Image", morphClosingImage)

cv2.waitKey(0)

File: c++ code/week1/examples/binary image processing/binary image processing/closing.cpp
#include <opencv2/opencv.hpp>

int main()
{
	// read image
	cv::Mat image = cv::imread("../../../../../images/closing.png", cv::IMREAD_GRAYSCALE);

	// initialize number of iteration
	int iterations = 1;

	// create a structuring kernel
	int closingSize = 10;
	cv::Mat element = cv::getStructuringElement(cv::MORPH_ELLIPSE, 
		cv::Size(2 * closingSize + 1, 2 * closingSize + 1), 
		cv::Point(closingSize, closingSize));

	// apply morphological closing
	cv::Mat morphClosedImage;
	cv::morphologyEx(image, morphClosedImage, cv::MORPH_CLOSE, element, cv::Point(-1, -1), iterations);

	// display images
	cv::imshow("Original Image", image);
	cv::imshow("Closing Image", morphClosedImage);

	cv::waitKey(0);

	return EXIT_SUCCESS;
}